### PR TITLE
Properly layer feature level code action tests

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This will prevent us from having to move features level tests to the editor feature layer.

### CODE REVIEW

File: src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.UnitTestUtilities.csproj [hunk 1]
```diff
@@ -9,14 +9,14 @@
     <IsShipping>false</IsShipping>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="..\..\..\EditorFeatures\DiagnosticsTestUtilities\CodeActions\AnalyzerProperty.cs" Link="AnalyzerProperty.cs" />
-    <Compile Include="..\..\..\EditorFeatures\DiagnosticsTestUtilities\CodeActions\CodeFixVerifierHelper.cs" Link="CodeFixVerifierHelper.cs" />
+    <Compile Include="..\..\..\Features\DiagnosticsTestUtilities\CodeActions\AnalyzerProperty.cs" Link="AnalyzerProperty.cs" />
+    <Compile Include="..\..\..\Features\DiagnosticsTestUtilities\CodeActions\CodeFixVerifierHelper.cs" Link="CodeFixVerifierHelper.cs" />
     <Compile Include="..\..\..\EditorFeatures\DiagnosticsTestUtilities\CodeActions\CSharpCodeFixVerifier`2+Test.cs" Link="CSharpCodeFixVerifier`2+Test.cs" />
     <Compile Include="..\..\..\EditorFeatures\DiagnosticsTestUtilities\CodeActions\CSharpCodeFixVerifier`2.cs" Link="CSharpCodeFixVerifier`2.cs" />
```
Comment (CyrusNajmabadi) [inline src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.UnitTestUtilities.csproj:15]: the goal is to move these as well.

File: src/Features/DiagnosticsTestUtilities/Microsoft.CodeAnalysis.Features.DiagnosticsTests.Utilities.csproj [hunk 2]
```diff
@@ -9,23 +9,6 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (CyrusNajmabadi) [inline src/Features/DiagnosticsTestUtilities/Microsoft.CodeAnalysis.Features.DiagnosticsTests.Utilities.csproj:9]: this project already was checked in, but not referenced in the Roslyn solution.  I updated it to have hte contents of the Editor version, without the actual references to any editor layers.

Comment (tmat) [reply src/Features/DiagnosticsTestUtilities/Microsoft.CodeAnalysis.Features.DiagnosticsTests.Utilities.csproj:9]: Ah, my bad. I think I ventured into this rabbit hole for a bit and then backed out, but did not clean up everything.

**Comment by CyrusNajmabadi** [comment]
The parts remaining to move down are:

image

I'm not doing that here as there is more work involved here, and there are legit parts that should likely stay at the editor level. for example, tests of things like dialogs and previews.  However, all of the stuff in this pr (which many tests are using) do not need any editor dependencies.

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**